### PR TITLE
[FEATURE] Add Deployer

### PR DESCRIPTION
**Pull request by brotkrueml**
Resolves: #204

### CODE REVIEW

File: Documentation/Installation/Deployer/Index.rst [hunk 1]
```diff
@@ -1,17 +1,90 @@
...
+    vendor/bin/dep init
+
+On console you are asked interactively some questions to generate a
+configuration:
+
+#.  Recipe language: The following sections use the PHP configuration, so choose
+    `php`.
+
+#.  Project template: You may choose `typo3` as project template, but this is
+    slightly outdated, so it is suggested to use the `common` template.
```
Comment (garvinhicking) [inline Documentation/Installation/Deployer/Index.rst:51]: Maybe add: "To better understand the deployment process and its configuration, and to be able to customize the process, choose the template 'common'"?

File: Documentation/Installation/Deployer/Index.rst [hunk 2]
```diff
@@ -1,17 +1,90 @@
...
+
+#.  Repository: Enter the repository URL of your TYPO3 project.
```
Comment (garvinhicking) [inline Documentation/Installation/Deployer/Index.rst:53]: This about a git repository? How would one enter the URL, using https:// or ssh://, an example might help to understand.
If auto-detection is used, maybe mention this here?

File: Documentation/Installation/Deployer/Index.rst [hunk 3]
```diff
@@ -1,17 +1,90 @@
...
+========================
+
+Afterwards you can initialize the configuration with
+
+..  code-block:: bash
+
+    vendor/bin/dep init
+
+On console you are asked interactively some questions to generate a
+configuration:
```
Comment (garvinhicking) [inline Documentation/Installation/Deployer/Index.rst:45]: "Deployer suggest some auto-detected defaults for some of the options"

File: Documentation/Installation/Deployer/Index.rst [hunk 4]
```diff
@@ -1,17 +1,90 @@
...
+#.  Project template: You may choose `typo3` as project template, but this is
+    slightly outdated, so it is suggested to use the `common` template.
+
+#.  Repository: Enter the repository URL of your TYPO3 project.
+
+#.  Project name: Enter the name of the project.
+
+#.  Hosts: Leave empty for now, we will come back later to this.
+
+A :file:`deploy.php` file has been created, which is the starting point:
```
Comment (garvinhicking) [inline Documentation/Installation/Deployer/Index.rst:59]: Also mention the created "recipe/common.php"? Or where is this required from?

File: Documentation/Installation/Deployer/Index.rst [hunk 5]
```diff
@@ -1,17 +1,90 @@
...
 
-..  literalinclude:: _deploy.php
+..  todo: Explain:
+    * releases (which are symlinked)
+
+.. contents:: **Table of Contents**
+   :local:
+
+
+Installation
```
Comment (garvinhicking) [inline Documentation/Installation/Deployer/Index.rst:23]: Maybe some preface here like:

Configurations created with Deployer are called "recipes", and they contain the steps to perform for your deployment.

A usual deployment performs these steps:

* Initialize configuration
* Check out the project's repository in a (temporary, project-independent) workspace
* Install the project dependencies (including TYPO3) with Composer
* Perform possible extra steps like running JavaScript frontend builds, fetching assets
* Collect all created files and transfer them to the target server, ensuring proper directory structure
* Execute remote commands on the target server (for example flushing the cache via the :file:`typo3` CLI binary

(Just a stub, not sure what I might be missing)

File: Documentation/Installation/Deployer/Index.rst [hunk 6]
```diff
@@ -1,17 +1,90 @@
...
+    common names are :file:`public`, :file:`htdocs` or :file:`web`. We will use
+    :file:`public` in the example.
+
+*   Some `shared_dirs` which hold content and must not change when switching
+    releases.
+
+*   Some `shared_files` may be necessary, if you use a .env file, this will
+    be referenced here.
+
+*   Some of the folders must be writable (`writable_dirs`). There a separate
```
Comment (garvinhicking) [inline Documentation/Installation/Deployer/Index.rst:84]:
```suggestion
*   Some of the folders must be writable (`writable_dirs`). There are separate
```

**Comment by brotkrueml** [comment]
Abandoned, as I will not pick this up in the next time.

**Comment by garvinhicking** [comment]
I'll pick this up then since it was mostly finished and a good addition

**Comment by brotkrueml** [comment]
> I'll pick this up then since it was mostly finished and a good addition

Nope, mostly the custom recipes are missing. The "typo3" recipe shipped with Deployer is outdated and cannot be used for current TYPO3 versions (and cannot be updated as it is "breaking").

For example: flushing and warming up cached.

Jochen created a Deployer task in Gitlab, maybe this is also a starting point:
https://gitlab.com/gitlab-org/project-templates/typo3-distribution/-/merge_requests/50

**Comment by linawolf** [comment]
I think this is not really a topic that belongs in getting started. I would, however, hate to lose the content.

It seems like we are missing a place to talk about deployment, advanced administration topics etc. Maybe it would be a good idea to start an administration guide? maybe the upgrade guide should actually be an administration, deployment and upgrade guide?

How would you feel about this @brotkrueml , @garvinhicking ?

**Comment by garvinhicking** [comment]
I'm not in favor of creating another guide, I would rather consolidate guides than create more of them. But I haven't looked into the structure of them for some time.

IMO we should discuss that seperately (because this affects other existinf content and not just this addition). Let's put it into the agenda for the next docs team meeting?

File: Documentation/Installation/Deployer/Index.rst [hunk 7]
```diff
@@ -1,17 +1,90 @@
...
+mainly the "Basics" and "Main Concepts".
 
-..  literalinclude:: _deploy.php
+..  todo: Explain:
+    * releases (which are symlinked)
+
+.. contents:: **Table of Contents**
+   :local:
+
+
```
Comment (linawolf) [inline Documentation/Installation/Deployer/Index.rst:22]:
```suggestion

.. _deployment-deployer-installion:
```

**Comment by brotkrueml** [comment]
I am in favour of closing this PR of mine. I started half a year ago at the community sprint and won't finish it. And Garvin hasn't taken over, yet. If there is a need for adding a Deployer section it can start from scratch or inspired by this PR.

**Comment by linawolf** [comment]
Ok, lets close it!

**Pull request closed without merging**